static/app/components/u2f/u2finterface.tsx

```diff
@@ -116,11 +121,18 @@ class U2fInterface extends Component<Props, State> {
           () => {
             const u2fResponse = this.getU2FResponse(data);
             const challenge = JSON.stringify(this.props.challengeData);
+            const authStateData = this.props.authStateData
+              ? JSON.stringify(this.props.authStateData)
+              : this.props.authStateData;
 
             if (this.state.responseElement) {
               // eslint-disable-next-line react/no-direct-mutation-state
               this.state.responseElement.value = u2fResponse;
             }
+            if (this.state.authStateElement && authStateData) {
+              // eslint-disable-next-line react/no-direct-mutation-state
+              this.state.authStateElement.value = authStateData;
```

Comment:
i'd really recommend against doing this directly manipulating state like this (it can cause a lot of difficult to detect issues).

instead take advantage of the react render life cycle around state updates, using `setState` to trigger changes checking for specific component conditions.